File: frontend/src/lib/models/page_elements/text_element_config.ts
import type { PageElementConfig } from "../page_element_config";
import type { Variable } from "../value_configs";

export let TEXT_ELEMENT_TYPE = "text";

export interface TextElementConfig extends PageElementConfig {
    value: Variable;
    options?: TextElementOptions;
}

export interface TextElementOptions {
    color?: string;
    font_size?: string;
}

File: src/main/kotlin/dev/schlaubi/dashboard/models/pageelements/TextElementConfig.kt
package dev.schlaubi.dashboard.models.pageelements

import dev.schlaubi.dashboard.models.PageVariable
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
@SerialName("text")
data class TextElementConfig(
    val value: PageVariable,
    val options: TextElementOptions? = null,
) : PageElementConfig()

@Serializable
data class TextElementOptions(
    val color: String? = null, @SerialName("font_size") val fontSize: String? = null
)